app/javascript/mastodon/features/direct_timeline/components/conversations_list.jsx

```diff
@@ -1,77 +1,70 @@
-import ConversationContainer from '../containers/conversation_container';
+import { expandConversations } from 'mastodon/actions/conversations';
+import ScrollableList from 'mastodon/components/scrollable_list';
 
-export default class ConversationsList extends ImmutablePureComponent {
+import { Conversation } from './conversation';
 
-  static propTypes = {
-    conversations: ImmutablePropTypes.list.isRequired,
-    scrollKey: PropTypes.string.isRequired,
-    hasMore: PropTypes.bool,
-    isLoading: PropTypes.bool,
-    onLoadMore: PropTypes.func,
-  };
-
-  getCurrentIndex = id => this.props.conversations.findIndex(x => x.get('id') === id);
+export const ConversationsList = ({ scrollKey, ...other }) => {
+  const listRef = useRef();
+  const conversations = useSelector(state => state.getIn(['conversations', 'items']));
+  const isLoading = useSelector(state => state.getIn(['conversations', 'isLoading'], true));
+  const hasMore = useSelector(state => state.getIn(['conversations', 'hasMore'], false));
+  const dispatch = useDispatch();
 
-  handleMoveUp = id => {
-    const elementIndex = this.getCurrentIndex(id) - 1;
-    this._selectChild(elementIndex, true);
-  };
+  const handleMoveUp = useCallback(id => {
+    const elementIndex = conversations.findIndex(x => x.get('id') === id) - 1;
+    focusChild(elementIndex, true);
+  }, [conversations]);
 
-  handleMoveDown = id => {
-    const elementIndex = this.getCurrentIndex(id) + 1;
-    this._selectChild(elementIndex, false);
-  };
+  const handleMoveDown = useCallback(id => {
+    const elementIndex = conversations.findIndex(x => x.get('id') === id) + 1;
+    focusChild(elementIndex, false);
+  }, [conversations]);
 
-  _selectChild (index, align_top) {
-    const container = this.node.node;
-    const element = container.querySelector(`article:nth-of-type(${index + 1}) .focusable`);
+  const focusChild = (index, alignTop) => {
+    const node = listRef.current;
+    const element = node.querySelector(`article:nth-of-type(${index + 1}) .focusable`);
 
     if (element) {
-      if (align_top && container.scrollTop > element.offsetTop) {
+      if (alignTop && node.scrollTop > element.offsetTop) {
         element.scrollIntoView(true);
-      } else if (!align_top && container.scrollTop + container.clientHeight < element.offsetTop + element.offsetHeight) {
+      } else if (!alignTop && node.scrollTop + node.clientHeight < element.offsetTop + element.offsetHeight) {
         element.scrollIntoView(false);
       }
+
       element.focus();
     }
-  }
-
-  setRef = c => {
-    this.node = c;
   };
 
-  handleLoadOlder = debounce(() => {
-    const last = this.props.conversations.last();
+  const handleLoadMore = debounce(() => {
+    const lastStatusId = conversations.last()?.get('last_status');
 
-    if (last && last.get('last_status')) {
-      this.props.onLoadMore(last.get('last_status'));
+    if (lastStatusId) {
+      dispatch(expandConversations({ lastStatusId }));
```

Comment:
The parameter name is incorrect, it should be maxId:
```suggestion
      dispatch(expandConversations({ maxId: lastStatusId }));
```